81407 - Apply code to maintain full asynchronous behavior to existing versions

OlyaAn opened the pull request:
[AB#81407](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/81407)

Commit:
81407 - Apply code to maintain full asynchronous behavior to existing versions

neexite commented on src/Orckestra.Composer.Cart/Api/CartController.cs:
```diff
@@ -144,7 +144,7 @@ public virtual async Task<IHttpActionResult> GetShippingMethods()
                 CultureInfo = ComposerContext.CultureInfo,
                 CustomerId = ComposerContext.CustomerId,
                 CartName = CartConfiguration.ShoppingCartName
-            });
```
Do you check another places? For example line 64 without ConfigureAwait?

OlyaAn replied:
@neexite updated controller

Commit:
updated CartController

neexite commented on src/Orckestra.Composer.Cart/Api/PaymentController.cs:
```diff
@@ -108,7 +108,7 @@ public async Task<IHttpActionResult> UpdateCheckoutPayment(UpdatePaymentMethodVi
 
-            var vm = await PaymentViewService.UpdateActivePaymentMethodAsync(param);
+            var vm = await PaymentViewService.UpdateActivePaymentMethodAsync(param).ConfigureAwait(false);
 
```
Line 258 for example still without .ConfigureAwait(false);

neexite replied:
I'm wondering why you keep GetActivePayment without ConfigureAwait? and other places

Commit:
updated PaymentController

Commit:
added ConfigureAwait

neexite commented on src/Orckestra.Composer.Cart/Services/CheckoutService.cs:
```diff
@@ -211,7 +211,7 @@ public virtual async Task<UpdateCartResultViewModel> UpdateCheckoutCartAsync(Upd
                 await operation.Invoke(cart);
```
I'm wondering why we do not use ConfigureAwait there.

Commit:
added configureAwait

neexite commented on src/Orckestra.Composer.Cart/Services/RecurringOrderCartsViewService.cs:
```diff
@@ -478,7 +478,7 @@ await CartRepository.RemoveCouponsAsync(new RemoveCouponsParam
+            var viewModel = await CreateCartViewModelAsync(vmParam).ConfigureAwait(false);
 
             return viewModel;
         }
```
line 548

Commit:
update

Commit:
update

neexite commented:
\src\Orckestra.Composer.Sitemap\EventHandlers\SitemapEventRegistrator.cs line 69, line 86

neexite commented:
src\Orckestra.Composer.CompositeC1\Services\Scheduler.cs line 48

Commit:
update

neexite commented:
Please rewrite all ".Result" to async methods where it is possible, for example.
src\Orckestra.Composer\Api\LocalizationController.cs : 30
src\Orckestra.Composer.Cart\Factory\CartViewModelFactory.cs : 
Where it is not possible use alternative sync method, 
example:
src\Orckestra.Composer.C1CMS.Queries\Providers\SearchQueryUrlProvider.cs:25
In place where it is not possible to rewrite to async  
.ConfigureAwait(false).GetAwaiter().GetResult() intead of .Result

https://learn.microsoft.com/en-us/archive/msdn-magazine/2015/july/async-programming-brownfield-async-development

Commit:
added .ConfigureAwait(false).GetAwaiter().GetResult()

Commit:
updated test

Commit:
fix

Commit:
fix

InnaBoitsun commented:
Found next issues:
**1. PageHead with error on Search Page** 
**2. Category Page**
![image](https://github.com/user-attachments/assets/6caf49fc-f9a0-4b9a-8fbc-7498ff0c1fff)

**3. Store Listing page:**
![image](https://github.com/user-attachments/assets/dfc64ba5-1faf-42ff-b725-8ba96cac59c7)

Commit:
fix

InnaBoitsun committed:
Clean redundant async () => await in cases when method returns the task directly without await, which is better in this case — avoids extra state machine generation.
Fix CacheProvider.GetOrAddAsyn inside async methods and add async() => await inside lambda - it is safer for deadlocks

#81407

InnaBoitsun committed:
clean redundant async() => await

InnaBoitsun reviewed (approved):
Looks ok. Need to merge and test website for possible issues. Then sync Grocery with changes

InnaBoitsun reviewed (changes requested):
@neexite please re-view also

neexite commented on src/Orckestra.Composer.CompositeC1/Controllers/BrowsingCategoriesBaseController.cs:
```diff
@@ -41,28 +42,28 @@ public BrowsingCategoriesBaseController(
         public virtual ActionResult ChildCategories(int page = 1, string sortBy = null, string sortDirection = null)
         {
-            return ExecuteBrowsing("ChildCategories", "ChildCategories", c => c, EmptyCategoryBrowsingContainer, page, sortBy, sortDirection);
+            return ExecuteBrowsingAsync("ChildCategories", "ChildCategories", c => c, EmptyCategoryBrowsingContainer, page, sortBy, sortDirection).ConfigureAwait(false).GetAwaiter().GetResult();
```
looks like this one easy to convert to async and do not use awaiter

InnaBoitsun replied:
I tried next:
```
 public virtual Task<ActionResult> ChildCategories(int page = 1, string sortBy = null, string sortDirection = null)
 {
     return ExecuteBrowsingAsync("ChildCategories", "ChildCategories", c => c, EmptyCategoryBrowsingContainer, page, sortBy, sortDirection);
 }
```
But this method "ChildCategories" registered as Function in C1 CMS and we have next error:
<img width="407" height="312" alt="image" src="https://github.com/user-attachments/assets/69b769dd-d89f-4a7d-bb92-f1dc5d0478ec" />

Next code works:
```
        public virtual ActionResult ChildCategories(int page = 1, string sortBy = null, string sortDirection = null)
        {
            return ExecuteBrowsingAsync("ChildCategories", "ChildCategories", c => c, EmptyCategoryBrowsingContainer, page, sortBy, sortDirection)
                .ConfigureAwait(false).GetAwaiter().GetResult();
        }
```

Please let me know what easy way you see. Thanks

InnaBoitsun committed:
more fixes

InnaBoitsun committed:
removed ConfigureAwait(false): is used to avoid resuming on the original synchronization context — which is helpful in async methods (e.g., await ...ConfigureAwait(false)).
But we are not awaiting this code —  .GetAwaiter().GetResult() to force sync execution.
Since the method is already blocking, ConfigureAwait(false) has no meaningful impact here.

InnaBoitsun committed:
return .ConfigureAwait(false)
